staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -641,6 +641,14 @@ func (r *crdHandler) removeDeadStorage() {
 			storageMap2[crd.UID] = storageMap[crd.UID]
 		}
 	}
+
+	// kcp: preserve partial metadata
+	for uid, crdInfo := range storageMap {
+		if strings.HasSuffix(string(uid), ".wildcard.partial-metadata") {
+			storageMap2[uid] = crdInfo
+		}
+	}
+
```

Comment:
do we have any means for cleanup? Or are we leaking now instead of deleting too much?

Reply:
Leaking. Need to fix that